xap-core/xap-datagrid/src/main/java/com/gigaspaces/internal/server/space/SpaceImpl.java

```
@@ -2629,9 +2630,14 @@ public GSEventRegistration notify(ITemplatePacket template, Transaction txn, lon
         }
     }
 
-    public void snapshot(ITemplatePacket template)
+    public void snapshot(ITemplatePacket template, SpaceContext sc)
```

Check if a client from previous build can connect to a Space with this change.
I think that because the LRMI methods were modified there will be a problem.
if so,  I would leave the public void snapshot(ITemplatePacket template) and add snapshot(ITemplatePacket template, SpaceContext sc)

---

@mavigdor I can leave both snapshot methods (with and without SpaceContext), but we should consider that user will be able to register new type in a secured space even without ALTER privilege. Below is a sample that I've created and it shows such behaviour.

`        //Invalid privilege validation via snapshot run if we don't check ALTER privilege in a snapshot
        SpaceTypeDescriptor typeDesc = new SpaceTypeDescriptorBuilder("Product2")
                .idProperty("id")
                .addFixedProperty("name", String.class)
                .addFixedProperty("price", Float.class)
                .addFixedProperty("id", Integer.class)
                .create();
        String[] _columnNames = typeDesc.getPropertiesNames();
        String[] _columnTypes = typeDesc.getPropertiesTypes();
        boolean[] _indices = typeDesc.getPropertiesIndexTypes();
        String _routingFieldName = typeDesc.getRoutingPropertyName();

        ISpaceProxy space = new GigaSpaceConfigurer(new SpaceProxyConfigurer(spaceName)
                .credentials("gs-admin", "gs-admin")) //gs-admin doesn't have ALTER (only READ) privilege to create new entry
                //but was able to create new entry via snapshot
                .gigaSpace()
                .getSpace()
                .getDirectProxy();
        ExternalEntry template = new ExternalEntry(typeDesc.getTypeName(), null, null);
        template.setFieldsNames(_columnNames);
        template.setFieldsTypes(_columnTypes);
        template.setIndexIndicators(_indices);
        template.setRoutingFieldName(_routingFieldName);
        SnapshotProxyActionInfo actionInfo = new SnapshotProxyActionInfo(space, template);

        space.getDirectProxy()
                .getRemoteJSpace()
                .snapshot(actionInfo.queryPacket);`

---

on the SpaceImpl if snapshot was called without sc and `isSecuredSpace` is true, we can throw an exception. This will avoid snapshot operations from clients that don't go through the correct path. Just keep the stub method so old clients can use the proxy. I think that if you remove it, there will be some serialization problem - worth checking, otherwise just remove it.